addons/l10n_in_gstin_status/models/res_partner.py

```diff
@@ -0,0 +1,105 @@
+    _inherit = 'res.partner'
+
+    l10n_in_gstin_verified_status = fields.Boolean(
+        string="GST Status",
+        tracking=True,
+    )
+    l10n_in_gstin_verified_date = fields.Date(
+        string="GSTIN Verified Date",
+        tracking=True,
+    )
+
+    @api.onchange('vat')
+    def _onchange_l10n_in_gst_status(self):
+        """
+        Reset GST Status Whenever the `vat` of partner changes
+        """
+        for partner in self:
+            if partner.country_code == 'IN':
+                partner.l10n_in_gstin_verified_status = False
+                partner.l10n_in_gstin_verified_date = False
+
+    def _l10n_in_gstin_status_get_url(self):
+        endpoint = self.env["ir.config_parameter"].sudo().get_param("l10n_in_gstin_status.endpoint", DEFAULT_IAP_ENDPOINT)
+        url = "%s%s" % (endpoint, '/iap/l10n_in_reports/1/public/search')
+        return url
+
+    def action_l10n_in_verify_gstin_status(self):
+        self.ensure_one()
+        if not self.vat:
+            raise ValidationError(_("Please enter the GSTIN"))
+        url = self._l10n_in_gstin_status_get_url()
+        params = {
+            "account_token": self.env["iap.account"].get("l10n_in_edi").account_token,
+            "dbuuid": self.env["ir.config_parameter"].sudo().get_param("database.uuid"),
+            "gstin_to_search": self.vat,
+        }
+        try:
+            response = jsonrpc(url, params=params, timeout=25)
+        except AccessError:
+            raise UserError(_("Unable to connect with GST network"))
+        if response.get('error') and any(e.get('code') == 'no-credit' for e in response['error']):
+            return self.env["bus.bus"]._sendone(self.env.user.partner_id, "iap_notification",
+                {
+                    "type": "no_credit",
+                    "title": _("Not enough credits to check GSTIN status"),
+                    "get_credits_url": self.env["iap.account"].get_credits_url(service_name=IAP_SERVICE_NAME),
+                },
+            )
+        gst_status = response.get('data', {}).get('sts', "")
+        if gst_status.casefold() == 'active':
+            l10n_in_gstin_verified_status = True
+        elif gst_status:
+            l10n_in_gstin_verified_status = False
+            date_from = response.get("data", {}).get("cxdt", '')
+            if date_from and re.search(r'\d', date_from):
+                message = _(
+                    "GSTIN %(vat)s is %(status)s and Effective from %(date_from)s",
+                    vat=self.vat,
+                    status=gst_status,
+                    date_from=date_from,
+                )
+            else:
+                message = _(
+                    "GSTIN %(vat)s is %(status)s, effective date is not available.",
+                    vat=self.vat,
+                    status=gst_status
+                )
+            self.message_post(body=message)
+        else:
+            _logger.info("GST status check error %s", response)
```

Comment:
![image](https://github.com/user-attachments/assets/c8a83d70-01a8-4324-837e-3a8d321255f5)

Should not we do a separate message for when the GST is invalid?  @hamo-odoo

Reply:
@jco-odoo Or just an error message and code
like `[SWEB_9035] Invalid GSTIN/UIN`

Reply:
if no error message and code then
`Unknown error while checking GST status`

Reply:
I still think the invalid GSTIN would deserve its own message as it could be a pretty common case, but the way you suggest is already way better.

Reply:
@jco-odoo okay,
@hamo-odoo wrote a good message for a special code `SWEB_9035` we get the benefit of translation in this case.